crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1663,6 +1663,87 @@ mod tests {
         assert_eq!(out_second_messages, vec![expected_c8y_json_message]);
     }
 
+    #[tokio::test]
+    async fn convert_measurement_for_child_device_service() {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, _http_proxy) = create_c8y_converter(&tmp_dir).await;
+
+        let in_topic = "te/device/child1/service/app1/m/m_type";
+        let in_payload = r#"{"temp": 1, "time": "2021-11-16T17:45:40.571760714+01:00"}"#;
+        let in_message = Message::new(&Topic::new_unchecked(in_topic), in_payload);
+
+        let expected_smart_rest_message_child = Message::new(
+            &Topic::new_unchecked("c8y/s/us"),
+            "101,test-device:device:child1,child1,thin-edge.io-child",
+        );
+        let expected_smart_rest_message_service = Message::new(
+            &Topic::new_unchecked("c8y/s/us/test-device:device:child1"),
+            "102,test-device:device:child1:service:app1,systemd,app1,up",
+        );
+        let expected_c8y_json_message = Message::new(
+            &Topic::new_unchecked("c8y/measurement/measurements/create"),
+            r#"{"externalSource":{"externalId":"test-device:device:child1:service:app1","type":"c8y_Serial"},"temp":{"temp":{"value":1.0}},"time":"2021-11-16T17:45:40.571760714+01:00","type":"m_type"}"#,
```

Comment:
I'd strongly recommend the `json!` macro to wrap such complex JSONs. I'd use it even for simpler ones with more than a couple of keys. I hope the macro doesn't change the order of keys so that your assertions work seamlessly, without needing `assert_json_matches!`

Reply:
fixed, using `json!` and `to_string` used, so don't need to use `assert_json_matches`.